### ISSUE TITLE
Google SSO with Interactive Embedding do not allow you to download CSV

### ISSUE THREAD

**Issue by Tony-metabase** [metabase/metabase#39848]
### Describe the bug

If you embed metabase in an iframe via Interactive Embedding and login via Google SSO you are not able to download questions and get an `Unauthorized` 401 error

<img width="1512" alt="image" src="https://github.com/metabase/metabase/assets/110378427/a32bc707-d075-4a70-bce4-e23c180a3bec">

### To Reproduce

You can use stats

1) Embed Stats in an iframe

```
<!DOCTYPE html>
<html lang="en">
<head>
    <iframe    src="https://stats-url.com"    allowtransparency></iframe>
</head>
<body>
</body>
</html>
```
2. Login via Google SSO
3. Try to download any question it will fail with an `Unauthorized` 401 error

Logging directly to Metabase via Google SSO and download a question works without isues:

<img width="1512" alt="image" src="https://github.com/metabase/metabase/assets/110378427/890ec9dc-b753-4c90-8177-6e290dd2e280">

### Expected behavior

I should be able to download a question since i was able to login inside the ifrme

### Logs

DEBUG metabase.server.middleware.log POST /api/dashboard/1880/dashcard/10011/card/11548/query/csv 401 128.0 µs (0 DB calls) 
"Unauthenticated"

### Information about your Metabase installation

```JSON
1.48.8 and master
```

### Severity

Blocking users from proceeding further since they use Google SSO internally and this blocks their full usage of metabase

**Comment by Tony-metabase** [comment]
@albertoperdomo is interactive embedding supposed to work with Google SSO?

**Comment by ArunavNobroker** [comment]
@Tony-metabase can we get this prioritised?

**Comment by WiNloSt** [comment]
@ArunavNobroker This should be released in 49.7 which is scheduled on 30th Apr.
